### PR TITLE
Improve several mod files for coreneuron in memory transfer.

### PR DESCRIPTION
**Pull request by WeinaJi**
Modify bbcore_read in `GlueSynapse.mod` and `StochKv.mod`.
Replace `VecStim.mod` with `vecevent.mod` from neuron repo.

### CODE REVIEW

File: common/mod/vecevent.mod [hunk 1]
```diff
@@ -0,0 +1,163 @@
+:  Vector stream of events, copied from nrn/share/examples/nrniv/netcon/vecevent.mod
+
+COMMENT
```
Comment (pramodk) [inline common/mod/vecevent.mod:3]: I guess this is just a copy?

File: common/mod/GluSynapse.mod [hunk 2]
```diff
@@ -668,8 +669,14 @@ static void bbcore_read(double* dArray, int* iArray, int* doffset, int* ioffset,
...
-        _p_delay_times = (double*)vector_new1(delay_times_sz);
-        _p_delay_weights = (double*)vector_new1(delay_weights_sz);
+        if (!_p_delay_times) {
+            _p_delay_times = (double*)vector_new1(delay_times_sz);
+        }
+        assert(delay_times_sz == vector_capacity((IvocVect*)_p_delay_times));
+        if (!_p_delay_weights) {
+            _p_delay_weights = (double*)vector_new1(delay_weights_sz);
+        }
+        assert(delay_weights_sz == vector_capacity((IvocVect*)_p_delay_weights));
```
Comment (pramodk) [inline common/mod/GluSynapse.mod:679]: Same pattern as other mod files, so should be good!

**Review by pramodk — APPROVED** [review]
Straightforward changes / same patterns for fixes!

LGTM! (of course, CI needs to be looked at!)

File: common/mod/vecevent.mod (hunk 1 above)
Comment (WeinaJi) [reply common/mod/vecevent.mod:3]: Yes, exactly copy of the `nrn/share/examples/nrniv/netcon/vecevent.mod`

**Comment by WeinaJi** [comment]
The CI is currently failing because we also have [VecStim.mod]( https://github.com/BlueBrain/neurodamus/blob/main/neurodamus/data/mod/VecStim.mod) in neurodamus, which is clearly not good. https://github.com/BlueBrain/neurodamus/pull/173 will remove `VecStim.mod` in neurodamus. 
A pipeline running with this branch and the relevant neurodamus branch has passed https://bbpgitlab.epfl.ch/hpc/sim/blueconfigs/-/pipelines/217403

**Review by pramodk — APPROVED** [review]

**Pull request merged**